src/models/AnalyticsGoal.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace DevGroup\Analytics\models;
+
+use Yii;
+
+/**
+ * This is the model class for table "analytics_goal".
+ *
+ * @property integer $id
+ * @property string $name
+ * @property integer $is_transactional
+ * @property integer $analytics_categories_id
+ * @property string $ga_action_name
+ * @property integer $ga_value
+ * @property string $ga_label
+ * @property string $ym_goal
+ * @property string $custom_params
+ * @property integer $once_per_session
+ * @property integer $once_per_visitor
+ *
+ * @property AnalyticsCategories $analyticsCategories
+ * @property VisitorSessionGoals $id0
+ * @property SelfReportingBlock[] $selfReportingBlocks
+ */
+class AnalyticsGoal extends \yii\db\ActiveRecord
+{
+    /**
+     * @inheritdoc
+     */
+    public static function tableName()
+    {
+        return '{{%analytics_goal}}';
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function rules()
+    {
+        return [
+            [['id', 'analytics_categories_id', 'ga_value'], 'required'],
+            [['id', 'is_transactional', 'analytics_categories_id', 'ga_value', 'once_per_session', 'once_per_visitor'], 'integer'],
+            [['name', 'ga_action_name', 'ga_label', 'ym_goal'], 'string'],
+            [['custom_params'], 'string', 'max' => 255],
```

Comment:
As custom params can be text - there's no limit on length, also it's meant to be encoded json params. Rename this attribute to `packed_json_params` and use [PackedJsonAttributes behavior](https://github.com/DevGroup-ru/yii2-data-structure-tools/blob/master/src/behaviors/PackedJsonAttributes.php)